tests/models/test_hooks.py

```diff
@@ -516,13 +516,9 @@ def training_step(self, batch, batch_idx):
         dict(name="setup", kwargs=dict(stage="fit")),
         dict(name="configure_sharded_model"),
         dict(name="Callback.on_configure_sharded_model", args=(trainer, model)),
-        # DeepSpeed skips initializing optimizers here as they are handled via config
-        *([dict(name="configure_optimizers")] if kwargs.get("strategy") != "deepspeed" else []),
+        dict(name="configure_optimizers"),
         dict(name="Callback.on_fit_start", args=(trainer, model)),
         dict(name="on_fit_start"),
-        # TODO: explore whether DeepSpeed can have the same flow for optimizers
-        # DeepSpeed did not find any optimizer in the config so they are loaded here
-        *([dict(name="configure_optimizers")] if kwargs.get("strategy") == "deepspeed" else []),
```

Comment:
Now, the hook order is consistent since there is no longer this problem: 

```py
setup()  # all plugins had optimizers setup here, except deepspeed
call_hook(on_fit_start)
pre_dispatch() # deepspeed had optimizers setup delayed until here
```

Now, setup and pre_dispatch is merged:

```py
setup()
call_hook(on_fit_start)
```